docs/example_go.md

```diff
@@ -3,29 +3,273 @@ title: "Example: Go"
+</figure>
+
+| Approach | Deploy Time |
+|---|---|
+| Naive | 4.2s |
+
+Can we do better?
+
+## Step 2: Let's Optimize for the Go Compiler
+
+When we make a change to a file, we currently copy all the source files to a
+container, download the dependencies, and re-compile the Go binary from scratch.
+
+But the Go compiler team has done a lot of work to make incremental compiles
+fast. How can we better use the Go tools how they're meant to be used?
+
+With `local_resource`, we can compile the Go binary locally, and copy the binary to
+the container.
+
+Here's our [new Tiltfile](https://github.com/windmilleng/tilt-example-go/blob/master/2-optimized/Tiltfile) 
+with the following new code:
+
+```python
+local_resource(
+  'example-go-compile',
+  'CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o build/tilt-example-go ./',
+  deps=['./main.go', './pkg'])
+
+docker_build(
+  'example-go-image',
+  '.',
+  dockerfile='deployments/Dockerfile',
+  only=[
+    './build',
+    './web',
+  ])
+```
+
+We've added a `local_resource()` that compiles the Go binary locally for our
+Linux container.
+
+We've added a new `only` parameter to `docker_build()`. This makes sure that the
+only directory in the Docker build context are the `./build` directory (for the
+compiled binary) and the `./web` directory (for the template and image files).
+
+We've also modified the Dockerfile to only copy the `./build` directory.
+
+Let's see what this looks like!
+
+<figure>
+  <a class="is-image" href="https://cloud.tilt.dev/snapshot/AcK4yOYLQObve-87cQ4=">
+    <img src="assets/docimg/example-go-image-3.png">
+  </a>
+  <figcaption>Step 2 complete. Click the screenshot to see an interactive snapshot.</figcaption>
+</figure>
+
+| Approach | Deploy Time |
+|---|---|
+| Naive | 4.2s |
+| Local Compile | 3.5s |
+
+
+## Step 3: Let's Optimize It
+
+When we make a change to a file, we currently have to build an image, deploy new Kubernetes configs,
+and wait for Kubernetes to schedule the pod.
+
+With Tilt, we can skip all of these steps, live-updating the pod in place.
+
+Here's our [new Tiltfile](https://github.com/windmilleng/tilt-example-html/blob/master/3-recommended/Tiltfile) 
+with the following new code:
+
+```python
+docker_build(
+  'example-go-image',
+  '.',
+  dockerfile='deployments/Dockerfile',
+  only=[
+    './build',
+    './web',
+  ],
+  live_update=[
+    sync('./build', '/app/build'),
+    sync('./web', '/app/web'),
+  ],
+  entrypoint = 'find . | entr -r ./build/tilt-example-go')
+```
+
+We've added a `live_update` parameter to `docker_build()` with two `sync` steps.
+They copy the code from the `./build` and `./web` directories into the container.
+
+We've also added a new parameter: `entrypoint="find . | entr -r ./build/tilt-example-go"`.
+
+`entr` is a tool that automatically restarts a shell command whenever the watched
+file changes. This command restarts our server every time the pod is updated.
+
+Congratulations on finishing this guide!
```

Comment:
This makes me wanna close the tab. Let's hold it until after the final score.

Reply:
hahaha, done